modules/apps/object/object-test/src/testFunctional/tests/ObjectNotifications.testcase

```diff
@@ -836,6 +836,125 @@ definition {
+
+		task ("And given a 1:M relationship between the Object A and Object B") {
+			ObjectAdmin.addObjectRelationshipViaAPI(
+				objectName_1 = "CustomObjectA182607",
+				objectName_2 = "CustomObjectB182607",
+				relationshipLabel = "Relationship",
+				relationshipName = "relationship",
+				relationshipType = "oneToMany");
+		}
+
+		task ("And given a custom User with admin roles is added") {
+			JSONUser.addUser(
+				userEmailAddress = "userea@liferay.com",
+				userFirstName = "userfn",
+				userLastName = "userln",
+				userScreenName = "usersn");
+
+			JSONRole.assignRoleToUser(
+				roleTitle = "Administrator",
+				userEmailAddress = "userea@liferay.com");
+		}
+
+		task ("And given a user notification template is created using terms from both objects") {
+			UserNotificationAPI.postNotificationTemplate(
+				en_US = "[%RELATIONSHIP_CUSTOMOBJECTA182607_CUSTOMFIELD%] | [%CUSTOMOBJECTB182607_CUSTOMFIELD%]",
+				externalReferenceCode = "ercTest",
+				name = "User Notification Template",
+				recipientType = "user",
+				type = "userNotification",
+				userScreenName = "usersn");
+		}
+
+		task ("And given the title field of Object A is changed") {
+			ObjectAdmin.openObjectAdmin();
+
+			ObjectPortlet.selectSystemObject(label = "Custom Object A 182607");
+
+			CreateObject.selectTitleField(fieldLabel = "Field Text A");
+
+			CreateObject.saveObject();
+		}
+
+		task ("And given a notification action is created on the Object B using the on after add trigger") {
+			Navigator.gotoBack();
+
+			ObjectPortlet.selectCustomObject(label = "Custom Object B 182607");
+
+			ObjectAdmin.goToActionsTab();
+
+			ObjectAdmin.addObjectActionViaUI(
+				actionLabel = "Notification action when entry is added",
+				active = "true",
+				notificationName = "User Notification Template",
+				objectLabel = "Custom Object B 182607",
+				thenAction = "Notification",
+				whenAction = "On After Add");
+		}
+
+		task ("And given the new custom User are logged") {
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "userea@liferay.com",
+				userLoginFullName = "userfn userln");
+		}
+
+		task ("When an entry in Object B is added") {
+			ObjectAdmin.addObjectSingleFieldEntryViaAPI(
+				fieldName = "customField",
+				objectName = "CustomObjectA182607",
+				value = "Entry Test A");
+
+			ObjectAdmin.goToCustomObject(objectName = "CustomObjectB182607");
+
+			LexiconEntry.gotoAdd();
+
+			Type(
+				key_fieldLabel = "Field Text B",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "Entry Test B");
+
+			ObjectPortlet.searchEntryRelationshipField(entryValue = "Entry Test A");
+
+			PortletEntry.save();
+		}
+
+		task ("Then an user notification is received with the terms replaced with the correct values") {
+			Notifications.notificationQueueAdmin();
+
+			AssertElementPresent(
+				key_objectName = "CustomObjectB182607",
+				locator1 = "ObjectNotifications#OBJECT_QUEUE_NAME");
+
+			AssertTextPresent(value1 = "Entry Test A | Entry Test B");
```

Comment:
The test is to assert the user notification is received and the terms are populated correctly. We shouldn't be checking the queue here.